src/main/java/de/unistuttgart/overworldbackend/service/NPCService.java

```diff
@@ -118,6 +127,9 @@ public NPCDTO updateNPCFromDungeon(
     final int npcIndex,
     final NPCDTO npcDTO
   ) {
+    List<PlayerNPCStatistic> statistics = playerNPCStatisticRepository.findByNpcId(npcDTO.getId());
+    statistics.forEach(statistic -> statistic.setCompleted(false));
+    playerNPCStatisticRepository.saveAll(statistics);
```

Comment:
I write this here also for the code snippet above: It should be better we write a update query instead of getting all statistics, change the value and save then all objects.

Reply:
Is it? I'm not quite sure, the performance gain is probably marginal.
And it doesn't really make the code too much clearer…